extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -404,20 +400,29 @@ export class AmpA4A extends AMP.BaseElement {
         })
         // This block returns true iff the creative was rendered in the shadow
         // DOM.
-        /** @return {!Promise<!boolean>} */
+        /** @return {!Promise<?CreativeMetaDataDef>} */
         .then(creative => {
           checkStillCurrent(promiseId);
-          // Note: It's critical that #maybeRenderAmpAd_ be called
+          // Note: It's critical that #getAmpAdMetadata_ be called
           // on precisely the same creative that was validated
           // via #validateAdResponse_.  See GitHub issue
           // https://github.com/ampproject/amphtml/issues/4187
 
           // TODO(levitzky) If creative comes back null, we should consider re-
           // fetching the signing server public keys and try the verification
           // step again.
-          return creative && this.maybeRenderAmpAd_(creative);
-        })
-        .catch(error => this.promiseErrorHandler_(error));
+
+          // Only return amp metadata if valid AMP creative.
+          if (!creative) {
+            return Promise.resolve();
+          }
+          // Need to know if creative was verified as part of render outside
+          // viewport but cannot wait on promise.  Sadly, need a state a
+          // variable.
+          this.isVerifiedAmpCreative_ = true;
+          return utf8Decode(creative).then(
+            creative => this.getAmpAdMetadata_(creative));
```

Comment:
Done though I have to check that creative and creativeDecoded exist as part of the return chain